src/services/navigationBar.ts

```diff
@@ -192,6 +193,11 @@ namespace ts.NavigationBar {
                             addTopLevelNodes((<Block>functionDeclaration.body).statements, topLevelNodes);
                         }
                         break;
+
+                    default:
+                        const childrens: Node[] = [];
+                        forEachChild(node, child => { childrens.push(child) });

```

Comment:
It does sort the nodes first. Although it looks like it would be equivalent to add nodes in arbitrary order and sort `topLevelNodes` at the end.